src/core/execute.c

```diff
@@ -2568,82 +2693,69 @@ static int acquire_credentials(
 
         /* First, load credentials off disk (or acquire via AF_UNIX socket) */
         HASHMAP_FOREACH(lc, context->load_credentials) {
-                ReadFullFileFlags flags = READ_FULL_FILE_SECURE|READ_FULL_FILE_FAIL_WHEN_LARGER;
-                _cleanup_(erase_and_freep) char *data = NULL;
-                _cleanup_free_ char *j = NULL, *bindname = NULL;
-                bool missing_ok = true;
-                const char *source;
-                size_t size, add;
+                int sub_fd;
```

Comment:
you are leaking this fd. Use `_cleanup_close_ int sub_fd = -1;` — that way the fd is automatically closed when this scope is left.